### ISSUE TITLE
Implement support for exporting Chapel arrays, part 2a - compiler changes

### ISSUE THREAD

**Issue by lydia-duncan** [chapel-lang/chapel#9603]
This is intended to cover the compiler changes necessary to support exported functions that take array arguments.  When exporting a function that takes an array argument, change resolution/some other pass so that it makes a separate function that gets exported and takes a C-style array, creates the array wrapper using the interface I'm writing in #9600 and calls the original function with that wrapper.

**Comment by lydia-duncan** [comment]
The check for if an exported function is generic occurs in FnSymbol::hasGenericFormals, which is called by FnSymbol::tagIfGeneric from resolve() before we start resolving extern vars and uses (which is prior to resolving the program itself).  It seems reasonable to add a function call prior to this call chain which checks for exported functions that take in array arguments

**Comment by lydia-duncan** [comment]
My original placement for this new function (at the beginning of resolve, prior to tagIfGeneric) is probably not going to work: the argument at this point only knows it is an _array, it doesn't know its declared element type (which I need in order to make the wrapper correctly).  Which means I need to perform this transformation when that type information is still around (probably normalize?  I'll check the log output)

I'm unclear how we forbid arrays of different types getting sent in as arguments when the type limitation is removed from the argument.

**Comment by lydia-duncan** [comment]
Ohhhh, it looks like we create a where clause for the function.  I was wondering what the last { } section was

**Comment by lydia-duncan** [comment]
Okay, so prior to normalize, the argument is declared with an exprType (on its DefExpr) involving the element type.  So as I suspected, I may need to do this work there, or postpone the tagIfGeneric check until later when an array is involved.

**Comment by lydia-duncan** [comment]
Finished with #9874
